src/admin/siteMessageAdmin/SystemMessagePreview.jsx

```diff
@@ -0,0 +1,101 @@
+import React, {useCallback, useState} from 'react'
+import SystemMessage from '../../systemMessage/SystemMessage.jsx'
+import routes from '../../app/routes.jsx'
+import TextField from '@mui/material/TextField'
+import MenuItem from '@mui/material/MenuItem'
+import Button from '@mui/material/Button'
+import {Collapse} from '@mui/material'
+
+export default function SystemMessagePreview() {
+
+    const allPages = [...new Set(routes.reduce((acc, route) => {
```

Comment:
This one can be either memoized or pulled outside the render function probably.

Reply:
moved out.